src/main/java/control/Parser.java

```diff
@@ -0,0 +1,49 @@
+package control;
```

Comment:
Many classes seem to be in a single package control? (at least thus far) think a bit about SOC and how you want to organize the code.